src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.ArrayList;
+
+
+import util.DukeException;
+
+public class Deadline extends Task {
+
+    protected String by;
+    static String divider = "    ═══*.·:·.☽✧    ✦    ✧☾.·:·.*═══";
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.taskType = "D";
+    }
+
+
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by.format(DateTimeFormatter.ofPattern("MMM d yyyy")).toString();
+    }
+
+    public Deadline(Boolean isDone, String description, String by) {
+        super(isDone, description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.by + ")";
+    }
+
+    
+
+    public static void createDeadlineTask(ArrayList<Task> array, String[] splitInput) {
+        if (splitInput.length == 1 || splitInput[1].equals("")){
+            try {
+                throw new DukeException("deadline");
+            } catch (Exception e) {
+                System.out.println(divider);
+                System.out.println(e.toString());
+                System.out.println(divider);
+            }
```

Comment:
I'm not too sure what this try-catch is trying to do though.

I'm assuming that you're expecting the throwing of a `DukeException` might fail?